proposals/53:kread-start/mn2-start.test.js

```diff
@@ -0,0 +1,441 @@
+// @ts-check
+
+/**
+ * @file mainnet-2 contract start test for KREAd
+ *
+ * Proposal info is fetched from a release.
+ *
+ * @typedef {{
+ *   bundles: string[],
+ *   evals: { permit: string; script: string }[],
+ * }} ProposalInfo
+ */
+
+import anyTest from 'ava';
+import * as cpAmbient from 'child_process'; // TODO: use execa
+import * as fspAmbient from 'fs/promises';
+import { tmpName as tmpNameAmbient } from 'tmp';
+import * as pathAmbient from 'path';
+import * as processAmbient from 'process';
+import dbOpenAmbient from 'better-sqlite3';
+
+// TODO: factor out ambient authority from these
+// or at least allow caller to supply authority.
+import {
+  agoric,
+  wellKnownIdentities,
+} from '../../upgrade-test-scripts/lib/cliHelper.js';
+import {
+  provisionSmartWallet,
+  voteLatestProposalAndWait,
+  waitForBlock,
+} from '../../upgrade-test-scripts/lib/commonUpgradeHelpers.js';
+
+import { makeAgd } from '../../upgrade-test-scripts/lib/agd-lib.js';
+import { dbTool } from '../../upgrade-test-scripts/lib/vat-status.js';
+import {
+  makeFileRW,
+  makeWebCache,
+  makeWebRd,
+} from '../../upgrade-test-scripts/lib/webAsset.js';
+import {
+  bundleDetail,
+  ensureISTForInstall,
+  flags,
+  getContractInfo,
+  loadedBundleIds,
+  testIncludes,
+  txAbbr,
+} from './core-eval-support.js';
+
+/** @typedef {Awaited<ReturnType<typeof makeTestContext>>} TestContext */
+/** @type {import('ava').TestFn<TestContext>}} */
+const test = anyTest;
+
+/**
+ * URLs of release assets, including bundle hashes agreed by BLD stakers
+ *
+ * TODO: get permits, scripts from blockchain?
+ * TODO: verify bundle contents against hashes?
+ *
+ * KREAd-rc1 to Mainnet
+ * voting 2023-09-28 to 2023-10-01
+ * https://agoric.explorers.guru/proposal/53
+ */
+const assetInfo = {
+  repo: {
+    release: 'https://github.com/Kryha/KREAd/releases/tag/KREAd-rc1',
+    url: 'https://github.com/Kryha/KREAd',
+    name: 'KREAd',
+  },
+  /** @type {Record<string, ProposalInfo>} */
+  buildAssets: {
+    'kread-committee-info.json': {
+      evals: [
+        {
+          permit: 'kread-invite-committee-permit.json',
+          script: 'kread-invite-committee.js',
+        },
+      ],
+      bundles: [
+        '/Users/wietzes/.agoric/cache/b1-51085a4ad4ac3448ccf039c0b54b41bd11e9367dfbd641deda38e614a7f647d7f1c0d34e55ba354d0331b1bf54c999fca911e6a796c90c30869f7fb8887b3024.json',
+        '/Users/wietzes/.agoric/cache/b1-a724453e7bfcaae1843be4532e18c1236c3d6d33bf6c44011f2966e155bc7149b904573014e583fdcde2b9cf2913cb8b337fc9daf79c59a38a37c99030fcf7dc.json',
+      ],
```

Comment:
I assume these values aren't read anywhere?
```suggestion
```

Reply:
Hm. What gives that impression? They're used in many places. all the calls to `bundleDetail`.

In particular, to fetch and install the bundles:

```js
      const { id, fileName, endoZipBase64Sha512 } = bundleDetail(bundle);
...
      const bundleRd = await assets.storedPath(fileName);
```

The `storedPath` should perhaps be called `provideStorePath`. It downloads the file if necessary.

Reply:
> What gives that impression? 

That they refer to paths that are only on Wietze's machine.

I've since learned that it parses out the `b1-….json`. I suggest omitting the leading path.